xls/dslx/stdlib/std.x

```diff
@@ -113,61 +113,67 @@ fn umul_test() {
   assert_eq(u4:0b1001, umul(u2:0b11, u2:0b11));
 }
 
-// Calculate x / y one bit at a time. This is an alternative to using
-// the division operator '/' which may not synthesize nicely.
-pub fn iterative_div<N: u32, DN: u32 = {N * u32:2}>(x: uN[N], y: uN[N]) -> uN[N] {
-  let init_shift_amount = ((N as uN[N])-uN[N]:1);
-  let x = x as uN[DN];
-
-  let (_, _, _, div_result) =
-  for (idx, (shifted_y, shifted_index_bit, running_product, running_result))
-      in range(u32:0, N) {
-
-    // Increment running_result by current power of 2
-    // if the prodcut running_result * y < x.
-    let inc_running_result = running_result | shifted_index_bit;
-    let inc_running_product = running_product + shifted_y;
-    let (running_result, running_product) = if inc_running_product <= x {
-      (inc_running_result, inc_running_product)
-    } else {
-      (running_result, running_product)
-    };
-
-    // Shift to next (lower) power of 2.
-    let shifted_y = shifted_y >> uN[N]:1;
-    let shifted_index_bit = shifted_index_bit >> uN[N]:1;
-
-    (shifted_y, shifted_index_bit, running_product, running_result)
-  } (( (y as uN[DN]) << (init_shift_amount as uN[DN]),
-       uN[N]:1 << init_shift_amount,
-       uN[DN]:0,
-       uN[N]:0));
-
-  div_result
+// Returns unsigned division of n (N bits) and d (M bits) as quotient (N bits) and remainder (M bits).
```

Comment:
Can we document the behavior of this implementation when dividing by zero?

If I'm not mistaken, as written, if d == 0, the output will be (q = all 1's, r = n).

Reply:
Done.